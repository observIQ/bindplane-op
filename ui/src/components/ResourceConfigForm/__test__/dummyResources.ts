import {
  ParameterDefinition,
  ParameterType,
  RelevantIfOperatorType,
  SourceType,
  Destination,
  PipelineType,
} from "../../../graphql/generated";
import { APIVersion } from "../../../types/resources";

/* -------------------------- ParameterDefinitions -------------------------- */

export const stringDef: ParameterDefinition = {
  name: "string_name",
  label: "String Input",
  description: "Here is the description.",
  required: false,

  type: ParameterType.String,

  default: "default-value",
};

export const stringDefRequired: ParameterDefinition = {
  name: "string_required_name",
  label: "String Input",
  description: "Here is the description.",
  required: true,

  type: ParameterType.String,

  default: "default-required-value",
};

export const enumDef: ParameterDefinition = {
  name: "enum_name",
  label: "Enum Input",
  description: "Here is the description.",
  required: false,

  type: ParameterType.Enum,

  default: "option1",
  validValues: ["option1", "option2", "option3"],
};

export const stringsDef: ParameterDefinition = {
  name: "strings_name",
  label: "Multi String Input",
  description: "Here is the description.",
  required: false,

  type: ParameterType.Strings,

  default: ["option1", "option2"],
};

export const boolDef: ParameterDefinition = {
  name: "bool_name",
  label: "Bool Input",
  description: "Here is the description.",
  required: false,

  type: ParameterType.Bool,

  default: true,
};

export const boolDefaultFalseDef: ParameterDefinition = {
  name: "bool_default_false_name",
  label: "Bool Default False Input",
  description: "Here is the description.",
  required: false,

  type: ParameterType.Bool,

  default: false,
};

export const intDef: ParameterDefinition = {
  name: "int_name",
  label: "Int Input",
  description: "Here is the description.",
  required: false,

  type: ParameterType.Int,

  default: 25,
};

export const relevantIfDef: ParameterDefinition = {
  name: "string_name",
  label: "String Input",
  description: "Here is the description.",
  required: false,

  type: ParameterType.String,

  relevantIf: [
    {
      name: "bool_default_false_name",
      operator: RelevantIfOperatorType.Equals,
      value: true,
    },
  ],

  default: "default-value",
};

/* ----------------------------- Resource Types ----------------------------- */

export const ResourceType1: SourceType = {
  apiVersion: APIVersion.V1_BETA,
  kind: "ResourceType",
  metadata: {
    id: "resource-type-1",
    name: "resource-type-1",
    displayName: "ResourceType One",
    description: "A description for resource one.",
    icon: "/icons/destinations/otlp.svg",
  },
  spec: {
    version: "0.0.0",
    parameters: [
      stringDef,
      stringDefRequired,
      enumDef,
      stringsDef,
      boolDef,
      intDef,
    ],
    telemetryTypes: [],

    supportedPlatforms: ["linux", "macos", "windows"],
  },
};

export const ResourceType2: SourceType = {
  apiVersion: APIVersion.V1_BETA,
  kind: "ResourceType",
  metadata: {
    id: "resource-type-2",
    name: "resource-type-2",
    displayName: "ResourceType Two",
    description: "A description for resource one.",
    icon: "/icons/destinations/otlp.svg",
  },
  spec: {
    version: "0.0.0",
    parameters: [boolDefaultFalseDef, relevantIfDef],

    supportedPlatforms: ["linux", "macos", "windows"],
    telemetryTypes: [],
  },
};

export const SupportsLogs: SourceType = {
  apiVersion: APIVersion.V1_BETA,
  kind: "ResourceType",
  metadata: {
    id: "supports-logs",
    name: "supports-logs",
    displayName: "Supports Logs",
    description: "A resource that supports logs.",
    icon: "/icons/destinations/otlp.svg",
  },
  spec: {
    version: "0.0.0",
    parameters: [boolDefaultFalseDef, relevantIfDef],

    supportedPlatforms: ["linux", "macos", "windows"],
    telemetryTypes: [PipelineType.Logs],
  },
};

export const SupportsMetrics: SourceType = {
  apiVersion: APIVersion.V1_BETA,
  kind: "ResourceType",
  metadata: {
    id: "supports-metrics",
    name: "supports-metrics",
    displayName: "Supports Metrics",
    description: "A resource that supports metrics.",
    icon: "/icons/destinations/otlp.svg",
  },
  spec: {
    version: "0.0.0",
    parameters: [boolDefaultFalseDef, relevantIfDef],

    supportedPlatforms: ["linux", "macos", "windows"],
    telemetryTypes: [PipelineType.Metrics],
  },
};

export const SupportsBoth: SourceType = {
  apiVersion: APIVersion.V1_BETA,
  kind: "ResourceType",
  metadata: {
    id: "supports-logs-and-metrics",
    name: "supports-logs-and-metrics",
    displayName: "Supports Logs and Metrics",
    description: "A resource that supports logs and metrics.",
    icon: "/icons/destinations/otlp.svg",
  },
  spec: {
    version: "0.0.0",
    parameters: [boolDefaultFalseDef, relevantIfDef],

    supportedPlatforms: ["linux", "macos", "windows"],
    telemetryTypes: [PipelineType.Logs, PipelineType.Metrics],
  },
};

/* -------------------------------- Resources ------------------------------- */

// This destination is type resource-type-1
export const Destination1: Destination = {
  apiVersion: APIVersion.V1_BETA,
  kind: "Destination",
  metadata: {
    name: "destination-1-name",
    id: "destination-1-name",
  },
  spec: {
    parameters: [],
    type: "resource-type-1",
  },
};

// This destination is type resource-type-1
export const Destination2: Destination = {
  apiVersion: APIVersion.V1_BETA,
  kind: "Destination",
  metadata: {
    name: "destination-2-name",
    id: "destination-2-name",
  },
  spec: {
    parameters: [],
    type: "resource-type-1",
  },
};
