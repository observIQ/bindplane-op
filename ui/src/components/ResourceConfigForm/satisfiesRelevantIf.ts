import { isEqual } from "lodash";
import { ParameterDefinition } from "../../graphql/generated";

export function satisfiesRelevantIf(
  formValues: { [name: string]: any },
  definition: ParameterDefinition
): boolean {
  if (definition.relevantIf == null) {
    return true;
  }

  const relevantIf = definition.relevantIf;

  for (const condition of relevantIf) {
    // TODO (dsvanlani) Right now we only support and expect the "equals" operator
    // Add a capability to satisfy other operators like "less than" or "greater than".
    if (!isEqual(formValues[condition.name], condition.value)) {
      return false;
    }
  }

  return true;
}
