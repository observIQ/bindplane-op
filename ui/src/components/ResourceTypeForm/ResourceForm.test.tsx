import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { ResourceConfigForm } from ".";
import {
  ParameterDefinition,
  ParameterType,
  RelevantIfOperatorType,
} from "../../graphql/generated";
import {
  ParameterInput,
  Tuple,
  tupleArrayToMap,
  valueToTupleArray,
} from "./ParameterInput";
import { satisfiesRelevantIf } from "./satisfiesRelevantIf";
import { ResourceType1, ResourceType2 } from "./__test__/dummyResources";
import renderer from "react-test-renderer";
import { type } from "os";

describe("satisfiesRelevantIf", () => {
  const formValues: { [key: string]: any } = {
    one: true,
    two: "bar",
    three: 25,
  };

  const param1: ParameterDefinition = {
    name: "string_name",
    label: "String Input",
    description: "Here is the description.",
    required: false,

    type: ParameterType.String,

    relevantIf: [
      {
        name: "one",
        operator: RelevantIfOperatorType.Equals,
        value: true,
      },
    ],
  };

  const param2: ParameterDefinition = {
    name: "string_name",
    label: "String Input",
    description: "Here is the description.",
    required: false,

    type: ParameterType.String,
    relevantIf: [
      {
        name: "one",
        operator: RelevantIfOperatorType.Equals,
        value: false,
      },
    ],

    default: "default-value",
  };

  it("param1 matches", () => {
    const got = satisfiesRelevantIf(formValues, param1);
    expect(got).toEqual(true);
  });
  it("param2 does not match", () => {
    const got = satisfiesRelevantIf(formValues, param2);
    expect(got).toEqual(false);
  });
});

describe("ResourceForm component", () => {
  it("does not display field if relevantIf isn't satisfied", () => {
    render(
      <ResourceConfigForm
        kind="destination"
        title={ResourceType2.metadata.displayName!}
        description={ResourceType2.metadata.description!}
        parameterDefinitions={ResourceType2.spec.parameters}
      />
    );
    const stringInput = screen.queryByText("String Input");
    expect(stringInput).toBeNull();
  });

  it("will render input when relevantIf is satisfied", () => {
    render(
      <ResourceConfigForm
        kind="destination"
        title={ResourceType2.metadata.displayName!}
        description={ResourceType2.metadata.description!}
        parameterDefinitions={ResourceType2.spec.parameters}
      />
    );
    let stringInput = screen.queryByLabelText("String Input");
    expect(stringInput).toBeNull();

    screen.getByRole("checkbox").click();
    stringInput = screen.getByLabelText("String Input");
    expect(stringInput).toBeInTheDocument();
  });

  it("maintains stateful formValues as correctType", async () => {
    const expectedValues = {
      name: "",
      string_name: "default-value",
      string_required_name: "default-required-value",
      enum_name: "option1",
      strings_name: ["option1", "option2"],
      int_name: 25,
      bool_name: true,
    };

    let saveDone = false;

    let values: { [key: string]: any } = {};
    function onSave(formValues: { [key: string]: any }) {
      values = Object.assign({}, formValues);
      saveDone = true;
    }
    render(
      <ResourceConfigForm
        onSave={onSave}
        kind="source"
        title={ResourceType1.metadata.displayName!}
        description={ResourceType1.metadata.description!}
        parameterDefinitions={ResourceType1.spec.parameters}
        includeNameField
      />
    );

    screen.getByText("Save").click();

    await waitFor(() => saveDone === true);
    expect(values).toEqual(expectedValues);
  });

  it("maintains stateful formValues as correctType after change", async () => {
    const expectedValues = {
      name: "",
      string_name: "default-value",
      string_required_name: "default-required-value",
      enum_name: "option1",
      strings_name: ["option1", "option2"],
      int_name: 50,
      bool_name: true,
    };

    let saveDone = false;

    let values: { [key: string]: any } = {};
    function onSave(formValues: { [key: string]: any }) {
      values = Object.assign({}, formValues);
      saveDone = true;
    }
    render(
      <ResourceConfigForm
        onSave={onSave}
        kind="source"
        title={ResourceType1.metadata.displayName!}
        description={ResourceType1.metadata.description!}
        parameterDefinitions={ResourceType1.spec.parameters}
        includeNameField
      />
    );

    fireEvent.change(screen.getByLabelText("Int Input"), {
      target: { value: 50 },
    });

    screen.getByText("Save").click();

    await waitFor(() => saveDone === true);
    expect(values).toEqual(expectedValues);
  });

  it("disables save button when name field has an error", async () => {
    render(
      <ResourceConfigForm
        onSave={() => {}}
        kind="destination"
        title={ResourceType1.metadata.displayName!}
        description={ResourceType1.metadata.description!}
        parameterDefinitions={ResourceType1.spec.parameters}
        includeNameField
      />
    );

    const nameField = screen.getByTestId("name-field");
    // this is an invalid name and should make the save button disabled
    fireEvent.change(nameField, { target: { value: "dest-" } });

    expect(screen.getByTestId("resource-form-save")).toBeDisabled();
  });
});

describe("MapParamInput", () => {
  it("valueToTupleArray", () => {
    const tests = [
      {
        value: {
          foo: "bar",
          blah: "baz",
        },
        expect: [
          ["foo", "bar"],
          ["blah", "baz"],
          ["", ""],
        ],
      },
      {
        value: null,
        expect: [["", ""]],
      },
    ];

    for (const test of tests) {
      const got = valueToTupleArray(test.value);
      expect(got).toEqual(test.expect);
    }
  });

  it("tupleArrayToMap", () => {
    const tests: { tuples: Tuple[]; expect: any }[] = [
      {
        tuples: [
          ["one", "two"],
          ["three", "four"],
        ],
        expect: {
          one: "two",
          three: "four",
        },
      },
      {
        tuples: [
          ["", "blah"],
          ["three", "four"],
          ["some", "thing"],
          ["", ""],
        ],
        expect: {
          three: "four",
          some: "thing",
        },
      },
      {
        tuples: [["", ""]],
        expect: {},
      },
    ];

    for (const test of tests) {
      const got = tupleArrayToMap(test.tuples);
      expect(got).toEqual(test.expect);
    }
  });

  it("renders correctly", () => {
    const mapParameter: ParameterDefinition = {
      required: true,
      label: "Label",
      description: "description",
      type: ParameterType.Map,
      default: {},
      name: "map_type_param",
    };
    const tree = renderer.create(<ParameterInput definition={mapParameter} />);
    expect(tree).toMatchSnapshot();
  });
});

describe("MultiEnumParameter", () => {
  it("renders correctly", () => {
    const multiEnumParameter: ParameterDefinition = {
      required: true,
      label: "Label",
      description: "description",
      type: ParameterType.MultiEnum,
      default: {},
      validValues: ["one", "two", "three", "four"],
      name: "multi_enum_type_param",
    };

    const tree = renderer.create(
      <ParameterInput definition={multiEnumParameter} />
    );
    expect(tree).toMatchSnapshot();
  });
});

describe("YamlParameter", () => {
  it("renders correctly", () => {
    const yamlParameter: ParameterDefinition = {
      required: true,
      label: "Label",
      description: "description",
      type: ParameterType.Yaml,
      default: "",
      name: "yaml_type_param",
    };

    const tree = renderer.create(<ParameterInput definition={yamlParameter} />);
    expect(tree).toMatchSnapshot();
  });
});
