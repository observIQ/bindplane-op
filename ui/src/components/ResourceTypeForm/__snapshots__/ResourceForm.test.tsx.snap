// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnumsParameter renders correctly 1`] = `
Array [
  <label
    className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
  >
    Label
  </label>,
  <p
    className="MuiFormHelperText-root css-1d1r5q-MuiFormHelperText-root"
  >
    description
  </p>,
  <div
    className="css-nen11g-MuiStack-root"
  >
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeSmall css-1ycxvqq-MuiSwitch-root"
      >
        <span
          className="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        one
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeSmall css-1ycxvqq-MuiSwitch-root"
      >
        <span
          className="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        two
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeSmall css-1ycxvqq-MuiSwitch-root"
      >
        <span
          className="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        three
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeSmall css-1ycxvqq-MuiSwitch-root"
      >
        <span
          className="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        four
      </span>
    </label>
  </div>,
]
`;

exports[`MapParamInput renders correctly 1`] = `
Array [
  <label
    aria-required={true}
    htmlFor="map_type_param"
  >
    Label
     *
  </label>,
  <p
    className="MuiFormHelperText-root css-1d1r5q-MuiFormHelperText-root"
  >
    description
  </p>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-1fshczy-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1ksu6kl-MuiTypography-root"
      >
        Key
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-13qb2yq-MuiTypography-root"
      >
        Value
      </p>
    </div>
  </div>,
  <div
    className="css-1okj3ks-MuiStack-root"
  >
    <div
      className="css-w4z10b-MuiStack-root"
    >
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        data-testid="map_type_param-0-0-input"
        onClick={[Function]}
      >
        <input
          className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-hdw1oc"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        data-testid="map_type_param-0-1-input"
        onClick={[Function]}
      >
        <input
          className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-hdw1oc"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
        data-testid="map_type_param-0-remove-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={18}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="3 6 5 6 21 6"
          />
          <path
            d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
          />
        </svg>
      </button>
    </div>
  </div>,
  <div
    className="MuiBox-root css-1um4smj"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          className="feather feather-plus-circle"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="12"
            x2="12"
            y1="8"
            y2="16"
          />
          <line
            x1="8"
            x2="16"
            y1="12"
            y2="12"
          />
        </svg>
      </span>
      New Row
    </button>
  </div>,
]
`;

exports[`YamlParameter renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
    htmlFor="yaml_type_param"
    style={
      Object {
        "backgroundColor": "#fff",
        "color": undefined,
        "padding": undefined,
      }
    }
  >
    Label
    <span
      aria-hidden={true}
      className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
    >
       
      *
    </span>
  </label>
  <div>
    <div
      className="w-tc-editor code-container expanded"
      style={
        Object {
          "backgroundColor": "#fff",
          "border": "1px solid #aaa",
          "boxSizing": "border-box",
          "fontFamily": "ui-monospace,SFMono-Regular,SF Mono,Consolas,Liberation Mono,Menlo,monospace",
          "fontSize": 12,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "left",
        }
      }
    >
      <textarea
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        className="w-tc-editor-text"
        data-testid="yaml-editor"
        name="yaml_type_param"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        required={true}
        spellCheck="false"
        style={
          Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "WebkitTextFillColor": "transparent",
            "background": "none",
            "border": 0,
            "boxSizing": "inherit",
            "color": "inherit",
            "display": "inherit",
            "fontFamily": "inherit",
            "fontSize": "inherit",
            "fontStyle": "inherit",
            "fontVariantLigatures": "inherit",
            "fontWeight": "inherit",
            "height": "100%",
            "left": 0,
            "letterSpacing": "inherit",
            "lineHeight": "inherit",
            "margin": 0,
            "minHeight": 200,
            "opacity": 0.8,
            "outline": 0,
            "overflow": "hidden",
            "overflowWrap": "break-word",
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
            "position": "absolute",
            "resize": "none",
            "tabSize": "inherit",
            "textIndent": "inherit",
            "textRendering": "inherit",
            "textTransform": "inherit",
            "top": 0,
            "whiteSpace": "pre-wrap",
            "width": "100%",
            "wordBreak": "keep-all",
          }
        }
        value=""
      />
      <div
        className="w-tc-editor-preview language-yaml"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<pre aria-hidden=\\"true\\"><code class=\\"code-highlight\\"></code><br></pre>",
          }
        }
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "inherit",
            "display": "inherit",
            "fontFamily": "inherit",
            "fontSize": "inherit",
            "fontStyle": "inherit",
            "fontVariantLigatures": "inherit",
            "fontWeight": "inherit",
            "letterSpacing": "inherit",
            "lineHeight": "inherit",
            "margin": 0,
            "minHeight": 200,
            "outline": 0,
            "overflowWrap": "break-word",
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
            "tabSize": "inherit",
            "textIndent": "inherit",
            "textRendering": "inherit",
            "textTransform": "inherit",
            "whiteSpace": "pre-wrap",
            "wordBreak": "keep-all",
          }
        }
      />
    </div>
  </div>
  <p
    className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained Mui-required css-1wc848c-MuiFormHelperText-root"
  >
    description
  </p>
</div>
`;
